### PR TITLE
73 update usability of trim function

### PR DESCRIPTION
**Pull request by KiraSqueak**
The trim functions in both core and slice have been updated to use the new functionality (that the user can choose an index to start timing from and end trimming to ex. 100, 200 would give the user the pixels between [100, 200]). I updated the names and the descriptions to make these a bit more obvious, please let me know if they make sense to you all.

### CODE REVIEW

File: src/core_ct/slice.py
Comment (TheSecondReal0) [inline src/core_ct/slice.py:39]: I'm not sure how useful this line is, since `trim()` is meant to be used for a wide variety of things. Maybe there should be a "potential use cases" section where this would fit better.

File: src/core_ct/core.py
Comment (TheSecondReal0) [inline src/core_ct/core.py:107]: I'm not sure I'm sold on left and right, since you could be trimming along the z axis where top and bottom would be more applicable

File: src/core_ct/slice.py
Comment (TheSecondReal0) [inline src/core_ct/slice.py:37]: Maybe this should say something more like "Create new slice containing trimmed data" or something. This is a nitpick, I might just be a little iffy on "create new by removing data"

File: src/core_ct/slice.py
Comment (TheSecondReal0) [inline src/core_ct/slice.py:42]: Maybe we should specify that the amount it trims symmetrically depends on `left_index`

**Review by TheSecondReal0 — CHANGES_REQUESTED** [review]
I think we might wanna let users only provide `left_index` or `right_index` to trim asymmetrically. Making it trim symmetrically should be more explicit in the code imo, like in this code snippet:
```python
my_slice = my_slice.trim(left_index=100)
```
I wouldn't really expect it to trim 100 pixels off both sides. Something like these would be more readable:
```python
my_slice = my_slice.trim(left_index=100, symmetric=True)
my_slice = my_slice.trim(symmetric=100)
```

I know this is completely contrary to some discussion we had during meetings and what these changes do but maybe trim should always take `x` pixels off a given side. I think we could implement `numpy` style indexing which would supersede this version of trim, allowing something like this:
```python
# trim a slice to only contain x indices 100-500
my_slice = my_slice[100:500]
# trim a core to only contain y indices 200-300
my_core = my_core[:, 200:300]
# take a slice at z=5, containing x indices 100-500 and y indices 200-300
new_slice = my_core[100:500, 200:300, 5]
```
This is definitely a topic for further discussion though and definitely not something to be implemented in this PR.

Other than that the PR looks good, it's definitely an improvement over what it is currently. I think before we merge we should demo it to Ryan to see what she thinks.

Also this should probably change the first version number, since it is breaking. We also might need to update the example notebook.

**Review by Atomator — APPROVED** [review]
Since all my changes have been addressed, I'm going to leave an approval on this so it can merged once @TheSecondReal0's comments have been resolved. That being said, we definitely need to make sure we update the example notebook before merging this.

**Comment by Atomator** [comment]
> Other than that the PR looks good, it's definitely an improvement over what it is currently. I think before we merge we should demo it to Ryan to see what she thinks.

I agree with this @TheSecondReal0. I think getting Ryan's opinion on it would be wise, and it also gives us an opportunity to talk with her about doing `numpy` style indexing as we decide what we want to do next.

**Comment by Atomator** [comment]
Functionality superseded by PR #93 (adding `numpy` style indexing)

**Pull request closed without merging**
